app/routes.py

```diff
@@ -1,2 +1,199 @@
+                "is_complete": False if not task.completed_at else True,
+            })
+        return jsonify(tasks_response)
+    elif request.method == "POST":
+        request_body = request.get_json()
+
+        # check for missing items
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+            return make_response({"details": "Invalid data"}, 400)
+
+        new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body["completed_at"])
+        db.session.add(new_task)
+        db.session.commit()
+
+        return make_response({
+            "task": {
+                "id": new_task.task_id,
+                "title": new_task.title,
+                "description": new_task.description,
+                "is_complete": False if not new_task.completed_at else True}}, 201)
+
+# Get one task and manipulate it
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("Task not found", 404)
+
+    if request.method == "GET":
+        return {"task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False if not task.completed_at else True
+                }}
+
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        # check for missing items
+        if "title" not in form_data or "description" not in form_data or "completed_at" not in form_data:
+            return make_response({"details": "Invalid data"}, 400)
+
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        task.completed_at = form_data["completed_at"]
+        db.session.commit()
+        return make_response({
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False if not task.completed_at else True}}, 200)
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return make_response({"details": f'Task {task.task_id} "{task.title}" successfully deleted'})
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_task_incomplete(task_id):
+    if request.method == "PATCH":
+        task = Task.query.get(task_id)
+        if task is None:
+            return make_response("", 404)
+
+        task.completed_at = None
+        db.session.commit()
+        return make_response({
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False if not task.completed_at else True}}, 200)
```

Comment:
great job here

- you could do the 404 check with .get_or_404 
- again you could do a helper function for the response so you don't have to retype it. You could also have the is_complete check in the helper function